tests/pipeline/test_term.py

```diff
@@ -330,6 +333,29 @@ class SomeFactorNoDType(SomeFactor):
         with self.assertRaises(InvalidDType):
             SomeFactor(dtype=1)
 
+    def test_latest_on_different_dtypes(self):
+        self.assertIsInstance(TestingDataSet.bool_col.latest, Filter)
+        self.assertIsInstance(TestingDataSet.float_col.latest, Factor)
+        self.assertIsInstance(TestingDataSet.datetime_col.latest, Factor)
+        self.assertIsInstance(TestingDataSet.int_col.latest, Factor)
+
+    def test_failure_timing_on_bad_missing_values(self):
```

Comment:
@llllllllll I think this addresses the concern you raised in person today.

I originally had made `Column` fail eagerly, but that was annoying because the error couldn't tell you the name of the column that failed.  The compromise here is to fail in `_BoundColumnDescr`, which means we still barf in class construction, but we can give you an error that includes the name of the bad column.